### PR TITLE
integration test pipeline

### PR DESCRIPTION
**Pull request by cmungall**
Addresses #68 

Currently throwing errors, e.g. #71

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,28 +41,63 @@ cob.owl: cob-edit.owl | build/robot.jar
...
+main_test: build/report.tsv cob.owl
 
-# totally as-hoc list for now
-COB_COMPLIANT = pato go cl obi uberon po uberon+cl ro envo  ogms doid hp chebi mp mondo
+# integration tests
+itest: itest_compliant itest_noncompliant
 
-itest: $(patsubst %, build/reasoned-%.owl, $(COB_COMPLIANT))
+itest_compliant: $(patsubst %, build/reasoned-%.owl, $(COB_COMPLIANT))
+itest_noncompliant: $(patsubst %, build/incoherent-%.owl, $(COB_COMPLIANT))
```
Comment (jamesaoverton) [inline Makefile:91]: Shouldn't this be `COB_NONCOMPLIANT`?

**Comment by jamesaoverton** [comment]
I'm glad to have this. I made got the consistent checks working but not most of the inconsistent ones.

Most of these comments are trivial:

- our top level is a mess; I'd like to create directories for `src/`, `tests/` and `views/`
- I'd like to move the testing code into `tests/Makefile`
- I don't understand all the temp file shuffling in your tasks: you don't trust Make to clean things up? You need to inspect failed build files?
- from a previous commit: why does `cob-base.owl` include external axioms? That's the opposite of what I expected
- `cob-to-external.tsv` could easily be a ROBOT template, no need for a Perl script

**Comment by cmungall** [comment]
Thanks, yes, let's move some things out of the TL in a separate PR. Do you have a favorite exemplar project layout?

> I don't understand all the temp file shuffling in your tasks: you don't trust Make to clean things up? You need to inspect failed build files?

Both, yes. Make could be killed with a `-9`, or the process may be suspended. Or we may have declared PRECIOUS. Given that many commands stream output (not robot, typically) there is a high chance that corrupt/incoherent output will cause downstream problems. Worse, this can be silent corruption. I have had experiences in large pipelines where truncated files have caused expensive errors.

> from a previous commit: why does cob-base.owl include external axioms? That's the opposite of what I expected

You mean the bridging axioms? As I see it these are owned by cob and should thus go in the base. 

> cob-to-external.tsv could easily be a ROBOT template, no need for a Perl script

The perl script is pure evil and should eventually die

**Comment by cmungall** [comment]
>  don't understand all the temp file shuffling in your tasks:

I assumed you meant the `>@.tmp && mv $@.tmp $@` idiom I use all the time

I'm not sure about the `touch $@.RUN` stuff i think I inherited that!

File: Makefile (hunk 1 above)
Comment (cmungall) [reply Makefile]: good catch, done

**Comment by cmungall** [comment]
going to merge now, as I want to get the new improved cob-to-external on master so others can start contributing

**Pull request merged**